crates/ln-dlc-node/src/node/payment_persister.rs

```
@@ -0,0 +1,92 @@
+    fn all(&self) -> Result<Vec<(PaymentHash, PaymentInfo)>>;
+}
+
+#[derive(Default, Clone)]
+pub struct PaymentMap(Arc<Mutex<HashMap<PaymentHash, PaymentInfo>>>);
+
+impl PaymentPersister for PaymentMap {
+    fn insert(&self, payment_hash: PaymentHash, info: PaymentInfo) -> Result<()> {
+        self.0.lock().unwrap().insert(payment_hash, info);
+
+        Ok(())
+    }
+
+    fn upsert(
+        &self,
+        payment_hash: &PaymentHash,
+        flow: PaymentFlow,
+        htlc_status: HTLCStatus,
+        preimage: Option<PaymentPreimage>,
+        secret: Option<PaymentSecret>,
+    ) -> Result<()> {
+        let mut payments = self.0.lock().unwrap();
+        match payments.get_mut(payment_hash) {
+            Some(mut payment) => {
+                payment.status = htlc_status;
+
+                if let Some(preimage) = preimage {
+                    payment.preimage = Some(preimage);
+                }
+
+                if let Some(secret) = secret {
+                    payment.secret = Some(secret);
+                }
+            }
+            None => {
+                payments.insert(
+                    *payment_hash,
+                    PaymentInfo {
+                        preimage,
+                        secret,
+                        status: htlc_status,
+                        amt_msat: MillisatAmount(None),
+                        flow,
+                        timestamp: OffsetDateTime::now_utc(),
+                    },
+                );
+            }
+        }
+
+        Ok(())
+    }
+
+    fn get(&self, payment_hash: &PaymentHash) -> Result<Option<(PaymentHash, PaymentInfo)>> {
+        let payments = self.0.lock().unwrap();
+        let info = payments.get(payment_hash);
+
+        let payment = info.map(|info| (*payment_hash, *info));
+
+        Ok(payment)
+    }
+
+    fn all(&self) -> Result<Vec<(PaymentHash, PaymentInfo)>> {
+        let payments = self.0.lock().unwrap();
```

same as above - can we return an Err?